mantidproject/mantid
# Use make_unique to create unique_ptrs

Resolves #14562 

Replaced `new` with `make_unique` where possible. I searched for:
- `unique_ptr` and `new` on the same line
- `unique_ptr` and `new` on adjacent lines
- `uptr` and `new` on the same line
- `uptr` and `new` on adjacent lines
- typedefs such as `SuccessorType` or `FactoryType` (didn't find any changes necessary here)

Also did similar searches with `reset`, but couldn't find anything to change there.

No need to update release notes.

**To test:**
Code review to ensure the changes are ok, and see if you can find anywhere I've missed. There are no functional changes in this PR and the existing tests should cover it. 
Installer available at `\\olympic\Babylon5\Scratch\TomPerkins\PR_15362_installers`.
